front-end/stateofthecards-front-end/src/components/MenuCard.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+import styles from "./MenuCard.module.css";
+import ReactCardFlip from "react-card-flip";
+
+interface IProps {
+	children: React.ReactElement[];
+	currentChild: number;
+	cssClass: string;
+}
+
+interface IState {
+	children: React.ReactElement[];
+	isFlipped: boolean;
+	frontChild: React.ReactElement;
+	backChild: React.ReactElement;
+	currentIndex: number;
+	cssClass: string;
+}
+
+class MenuCard extends Component<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+
+		this.state = {
+			children: props.children,
+			isFlipped: false,
+			frontChild: props.children[props.currentChild],
+			backChild:
+				props.children[
+					(props.currentChild + 1) % props.children.length
+				],
+			currentIndex: props.currentChild,
+			cssClass: props.cssClass,
+		};
+	}
+
+	setCurrentChild(index: number) {
```

Comment:
Don't manage which child is shown in state in `MenuCard`, manage it in the parent component using props.